src/uint256.cpp

```diff
@@ -156,3 +160,35 @@ uint64_t uint256::GetHash(const uint256& salt) const
 
     return ((((uint64_t)b) << 32) | c);
 }
+
+uint64_t CalculateHash(const uint32_t* const src, size_t length, const uint32_t* const salt)
+{
+    uint32_t a, b, c;
+    const uint32_t *pn = (const uint32_t*)src;
+    const uint32_t *salt_pn = (const uint32_t*)salt;
+    a = b = c = 0xdeadbeef + length;
+
+    while(length > 3 ) {
+        a += pn[0] ^ salt_pn[0];
+        b += pn[1] ^ salt_pn[1];
+        c += pn[2] ^ salt_pn[2];
+        HashMix(a, b, c);
+
+        length -= 3;
+        pn += 3;
+        salt_pn += 3;
+    }
+
+    switch(length) {
```

Comment:
transform the switch without "breaks" into something more readable. Unit test this comparing results with GetHash where possible